ssl/ssl_lib.c

```diff
@@ -1583,8 +1583,10 @@ void SSL_set_bio(SSL *s, BIO *rbio, BIO *wbio)
      * If the two arguments are equal then one fewer reference is granted by the
      * caller than we want to take
      */
-    if (rbio != NULL && rbio == wbio)
-        BIO_up_ref(rbio);
+    if (rbio != NULL && rbio == wbio) {
+        if (!BIO_up_ref(rbio))
+            return;
```

Comment:
I fit ever gets here this is a silent failure..

Reply:
Yes, is there a better alternative?

Reply:
`SSL_set_bio()` is part of ssl interface. the only options we have here are:
  - put assert here and let code crash/abort in debug mode
  - make code to crash even in release build (it's still valid option with failing up ref code is getting to undefined state)
  -  just return like we do. application will see some failures soon because following BIO operation will fail, reasonably written app will exit
  - change API

from those options we have putting assert there to check up ref is successful is the only realistic option I can think of.

Reply:
IMO it is good as is.